src/runners/ast_json_printer.ml

```diff
@@ -0,0 +1,430 @@
+            ("name", ident_to_json i) ;
+            ("variable_type", t_json);
+            ("expression", expr_annot_to_json e);
+        ]
+    | ParsedSyntax.LibTyp (i, ctr_defs) ->
+        `Assoc [
+            ("name", ident_to_json i) ;
+            ("node_type", `String "LibraryType");
+            ("ctr_defs", `List (List.map ~f:ctr_def_to_json ctr_defs));
+        ]
+
+let library_to_json (l : ParsedSyntax.library) =
+    let node_type_json = ("node_type", `String "Library") in
+    let lname_json = ("library_name", ident_to_json l.lname) in
+    let lentries_json = ("library_entries", `List(List.map ~f:lib_entry_to_json l.lentries)) in
+    `Assoc [lname_json; lentries_json; node_type_json]
+
+let elibrary_to_json (name, alias) =
+    let node_type_json = ("node_type", `String "ExternalLibrary") in
+    let name_json = ("name", ident_to_json name) in
+    let a_json = match alias with
+        |  Some a' -> ("alias", ident_to_json a')
+        |  None -> ("alias", `Null) in
+    `Assoc [name_json; a_json; node_type_json]
+
+let cmodule_to_json (cmod : ParsedSyntax.cmodule) =
+    let smver_json = ("scilla_major_version", `Int cmod.smver) in
+    let cname_json = ("name", ident_to_json cmod.cname) in
+    let libs_json = match cmod.libs with
+        | Some l -> ("library", library_to_json l)
+        | None  -> ("library", `Null) in
+    let elibs_json = ("external_libraries", `List (List.map ~f:elibrary_to_json cmod.elibs)) in
+    let contr_json = ("contract", contract_to_json cmod.contr) in
+    let node_type_json = ("node_type", `String "ContractModule") in
+    `Assoc [smver_json; cname_json; libs_json; elibs_json; contr_json; node_type_json]
+
+let lmodule_to_json (lmod : ParsedSyntax.lmodule) =
+    let libs_json = ("library", library_to_json lmod.libs) in
+    let elibs_json = ("external_libraries", `List (List.map ~f:elibrary_to_json lmod.elibs)) in
+    let node_type_json = ("node_type", `String "LibraryModule") in
+    `Assoc [libs_json; elibs_json; node_type_json]
+
+let scillalib_to_json (file) =
+    let r = check_parsing file ScillaParser.Incremental.lmodule in
+    match r with
+        | Error e -> fatal_error_gas e (Stdint.Uint64.of_int 0)
+        | Ok lmod ->
+            let mod_json =  lmodule_to_json lmod in
+            pout (sprintf "%s\n" (Yojson.Basic.pretty_to_string mod_json))
+
+
+let scilla_to_json (file) =
+    let r = check_parsing file ScillaParser.Incremental.cmodule in
+    match r with
+        | Error e -> fatal_error_gas e (Stdint.Uint64.of_int 0)
+        | Ok cmod -> 
+            let mod_json =  cmodule_to_json cmod in
+            pout (sprintf "%s\n" (Yojson.Basic.pretty_to_string mod_json))
+
+let () =
+    let cli = parse_cli () in
+    let f_ext = FilePath.get_extension cli.input_file in
+    match f_ext with
+        | "scillib" -> scillalib_to_json cli.input_file
+        | "scilla" -> scilla_to_json cli.input_file
+        | _ -> pout(sprintf "Unhandled type %s\n" f_ext)
```

Comment:
Please use the word "type" carefully when dealing with Scilla. The word to use here is "file extension".

Reply:
fixed in 30543ef